simplify op name determination into a single forward pass

dagitses opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64261

Note that this does not preserve byte-for-byte compatibility with
existing names.

Test Plan:
* Rely on CI to catch gross errors.
* Merge after release cut to catch subtle issues.

Differential Revision: [D30700647](https://our.internmc.facebook.com/intern/diff/D30700647)

Commit:
simplify op name determination into a single forward pass

Note that this does not preserve byte-for-byte compatibility with
existing names.

Test Plan:
* Rely on CI to catch gross errors.
* Merge after release cut to catch subtle issues.

[ghstack-poisoned]

Commit:
Update on "simplify op name determination into a single forward pass"


Note that this does not preserve byte-for-byte compatibility with
existing names.

Test Plan:
* Rely on CI to catch gross errors.
* Merge after release cut to catch subtle issues.

[ghstack-poisoned]

Commit:
Update on "simplify op name determination into a single forward pass"


Note that this does not preserve byte-for-byte compatibility with
existing names.

Test Plan:
* Rely on CI to catch gross errors.
* Merge after release cut to catch subtle issues.

[ghstack-poisoned]

Commit:
Update on "simplify op name determination into a single forward pass"


Note that this does not preserve byte-for-byte compatibility with
existing names.

Test Plan:
* Rely on CI to catch gross errors.
* Merge after release cut to catch subtle issues.

[ghstack-poisoned]

Commit:
Update on "simplify op name determination into a single forward pass"


Note that this does not preserve byte-for-byte compatibility with
existing names.

Test Plan:
* Rely on CI to catch gross errors.
* Merge after release cut to catch subtle issues.

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30700647).

albanD commented on tools/autograd/load_derivatives.py:
```diff
@@ -43,32 +43,15 @@ def load_derivatives(derivatives_yaml_path: str, native_yaml_path: str) -> Seque
 
+        # Keep track of how many of which ops we've seen so we can
+        # disambiguate them with a numeric suffix.
+        op_prefixes = Counter[str]()
```
Shouldn't this be `op_postfixes` ?

dagitses replied:
It's kinda both. The keys are the prefixes and the values are the suffixes. Thoughts for a better name?

albanD replied:
prefix_to_suffix_map ?

dagitses replied:
how about `op_counter`? i think that's even clearer.

albanD reviewed (approved):
LGTM except the naming of prefix vs postfix

Commit:
Update on "simplify op name determination into a single forward pass"


Note that this does not preserve byte-for-byte compatibility with
existing names.

Test Plan:
* Rely on CI to catch gross errors.
* Merge after release cut to catch subtle issues.

Differential Revision: [D30700647](https://our.internmc.facebook.com/intern/diff/D30700647)

[ghstack-poisoned]
